experimenter/experimenter/experiments/constants.py

```diff
@@ -280,6 +280,29 @@ def kinto_collections(self) -> set[str]:
 ]
 
 
+class ApplicationNameMap(models.TextChoices):
+    DESKTOP = (APPLICATION_CONFIG_DESKTOP.slug, APPLICATION_CONFIG_DESKTOP.app_name)
+    FENIX = (APPLICATION_CONFIG_FENIX.slug, APPLICATION_CONFIG_FENIX.app_name)
+    IOS = (APPLICATION_CONFIG_IOS.slug, APPLICATION_CONFIG_IOS.app_name)
+    FOCUS_ANDROID = (
+        APPLICATION_CONFIG_FOCUS_ANDROID.slug,
+        APPLICATION_CONFIG_FOCUS_ANDROID.app_name,
+    )
+    KLAR_ANDROID = (
+        APPLICATION_CONFIG_KLAR_ANDROID.slug,
+        APPLICATION_CONFIG_KLAR_ANDROID.app_name,
+    )
+    FOCUS_IOS = (APPLICATION_CONFIG_FOCUS_IOS.slug, APPLICATION_CONFIG_FOCUS_IOS.app_name)
+    KLAR_IOS = (APPLICATION_CONFIG_KLAR_IOS.slug, APPLICATION_CONFIG_KLAR_IOS.app_name)
+    MONITOR = (
+        APPLICATION_CONFIG_MONITOR_WEB.slug,
+        APPLICATION_CONFIG_MONITOR_WEB.app_name,
+    )
+    VPN = (APPLICATION_CONFIG_VPN_WEB.slug, APPLICATION_CONFIG_VPN_WEB.app_name)
+    FXA = (APPLICATION_CONFIG_FXA_WEB.slug, APPLICATION_CONFIG_FXA_WEB.app_name)
+    DEMO_APP = (APPLICATION_CONFIG_DEMO_APP.slug, APPLICATION_CONFIG_DEMO_APP.app_name)
```

Comment:
Can we not just dynamically derive this?

```
ApplicationNameMap = models.TextChoices([(a.slug, a.app_name) for a in .....])
```

Reply:
Sure, I was just following the example of the Application class. I'll update this though.

Reply:
Oh wait, the issue here is that each one is derived from a different ApplicationConfig. There is no list of all of them, so either I can keep it the way it is, or create a new list of all of them. Preference?

Reply:
Ya it's here:

https://github.com/mozilla/experimenter/blob/main/experimenter/experimenter/experiments/constants.py#L382-L394